26-dracula-ant/technical_specification.md

```diff
@@ -0,0 +1,132 @@
+# Implementation of a work package service for CLI client authentication (Dracula Ant)
+
+**Epic Type:** Implementation Epic
+  
+**Attention: Please do not put any confidential content here.**
+
+## Scope:
+
+We want to implement the work package service outlined in
+https://docs.ghga-dev.de/main/architecture_concepts/ac001_file_validation_and_encryption.html#client-configuration-and-authorization
+
+## Implementation Details:
+
+### REST API
+
+The Work Package Service REST API should have the following end points:
+
+Used by the web frontend to create work packages:
+
+- `POST /work-packages`
+  - auth header: internal access token
+  - request body:
+    - `dataset_id`: str (the ID of a dataset)
+    - `type`: enum (download/upload)
+    - `file_ids`: array of strings  (empty = all files of the dataset)
+    - `public_key`: string (the user's public Crypt4GH key)
+  - response body:
+    - `id`: str (the work package ID)
+    - `token`: str (encrypted work package access token)
+
+Used by the CLI client to retrieve work packages and create work order tokens:
+
+- `GET /work-packages/{work_package_id}`
+  - auth header: work package access token
+  - gets the specified work package
+- `POST /work-packages/{work_package_id}/file/{file_id}/work-order-tokens`
+  - auth header: work package access token
+  - gets an encrypted work order token for the specified work package and file
+
+The following endpoints could be added later to manage work packages in the web frontend,
+but for now, this will *not* be implemented and they are not part of the API.
+
+- `GET /work-packages`
+  - auth header: internal access token
+  - gets all list of all work packages of the current user
+- `GET /work-packages/{work_package_id}`
+  - auth header: internal access token
+  - gets work package with given id if it belongs to the current user
+- `DELETE /work-packages/{work_package_id}`
+  - auth header: internal access token
+  - deletes work package with given id if it belongs to the current user
```

Comment:
```suggestion
```

Reply:
This is not part of the spec, just a suggestion, but I have now added a note how deletion should be handled.